vertical-pod-autoscaler/pkg/utils/vpa/limit_and_request_scaling.go

```diff
@@ -133,19 +133,21 @@ func scaleQuantityProportionallyCPU(scaledQuantity, scaleBase, scaleResult *reso
 	originalMilli := big.NewInt(scaledQuantity.MilliValue())
 	scaleBaseMilli := big.NewInt(scaleBase.MilliValue())
 	scaleResultMilli := big.NewInt(scaleResult.MilliValue())
-	var scaledOriginal big.Int
-	scaledOriginal.Mul(originalMilli, scaleResultMilli)
-	scaledOriginal.Div(&scaledOriginal, scaleBaseMilli)
-	if scaledOriginal.IsInt64() {
-		result := resource.NewMilliQuantity(scaledOriginal.Int64(), scaledQuantity.Format)
-		if rounding == roundUpToFullUnit {
-			result.RoundUp(resource.Scale(0))
-		}
-		if rounding == roundDownToFullUnit {
-			result.Sub(*resource.NewMilliQuantity(999, result.Format))
-			result.RoundUp(resource.Scale(0))
+
+	var result big.Int
+	result.Mul(originalMilli, scaleResultMilli)
+	// If the division produces a remainder:
+	// - with roundUpToFullUnit, we apply ceiling to the value
+	// - with noRounding or roundDownToFullUnit, we apply floor to the value
+	// TODO(iamzili) - I think we eventually want to get rid of the noRounding mode.
+	z, m := new(big.Int).DivMod(&result, scaleBaseMilli, new(big.Int))
+	if z.IsInt64() {
+		if m.Sign() != 0 {
+			if rounding == roundUpToFullUnit {
```

Comment:
another nit:
can we combine these to save an extra nest level?
```suggestion
		if m.Sign() != 0 && rounding == roundUpToFullUnit {
```